src/__tests__/autoDetectRenderer.test.ts

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { ReactHooksRenderer } from '../types/react'
+
+describe('auto-detect renderer', () => {
```

Comment:
This test file replaces the `defaultRenderer` and `defaultRenderer.pure` test files.